physics/forkable_body.hpp

```diff
@@ -285,7 +283,8 @@ LowerBound(Instant const& time) const {
             // We found an interesting timeline, i.e. one that is nonempty and
             // not forked at the fork point of its parent.  Cut the ancestry and
             // return the beginning of that timeline.
-            iterator.ancestry_.erase(iterator.ancestry_.begin(), ancestry_it);
+            iterator.ancestry_ = {ancestry_it, iterator.ancestry_.rend()};
```

Comment:
This seems wrong.  The construction using reverse iterators is going to effectively reverse the range, so the element at `rend-1`, i.e., at `begin` will land at the back.  I would stick to `erase` here.  It would be good to add a test covering this situation: you'll need a trajectory with many forks, call `LowerBound` with a time near the middle, and check that the resulting iterator gives you data in the right order.

Reply:
You are correct; it is wrong. I added a new test and changed it to erase.